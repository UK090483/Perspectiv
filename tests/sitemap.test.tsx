import { render } from "@testing-library/react";
import { getServerSideProps } from "../pages/sitemap.xml";

jest.mock("@lib/SanityService/sanity.server", () => ({
  fetch: () => Promise.resolve([{ slug: "test1" }, { slug: "test2" }]),
}));

describe("Sitemap", () => {
  it("should render ", () => {
    expect(getServerSideProps()).toMatchSnapshot();
  });
});
