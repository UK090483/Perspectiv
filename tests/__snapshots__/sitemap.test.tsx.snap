// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sitemap should render  1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
    <urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\">
      
            <url>
              <loc>testSlug1</loc>
              <lastmod>2020-03-31T22:00:00.000Z</lastmod>
              <changefreq>monthly</changefreq>
              <priority>1.0</priority>
            </url>
          
            <url>
              <loc>testSlug2</loc>
              <lastmod>2020-03-31T22:00:00.000Z</lastmod>
              <changefreq>monthly</changefreq>
              <priority>1.0</priority>
            </url>
          
    </urlset>
  "
`;

exports[`Sitemap should render no Lang  1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
    <urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\">
      
            <url>
              <loc>testSlug1</loc>
              <lastmod>2020-03-31T22:00:00.000Z</lastmod>
              <changefreq>monthly</changefreq>
              <priority>1.0</priority>
            </url>
          
            <url>
              <loc>testSlug2</loc>
              <lastmod>2020-03-31T22:00:00.000Z</lastmod>
              <changefreq>monthly</changefreq>
              <priority>1.0</priority>
            </url>
          
    </urlset>
  "
`;

exports[`Sitemap should render with Lang  1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
    <urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\">
      
            <url>
              <loc>testSlug1</loc>
              <lastmod>2020-03-31T22:00:00.000Z</lastmod>
              <changefreq>monthly</changefreq>
              <priority>1.0</priority>
            </url>
          
            <url>
              <loc>testSlug2</loc>
              <lastmod>2020-03-31T22:00:00.000Z</lastmod>
              <changefreq>monthly</changefreq>
              <priority>1.0</priority>
            </url>
          
    </urlset>
  "
`;
